source/extensions/filters/common/ext_authz/ext_authz_grpc_impl.cc

```diff
@@ -57,6 +57,11 @@ void GrpcClientImpl::onSuccess(std::unique_ptr<envoy::service::auth::v3::CheckRe
     authz_response->status = CheckStatus::OK;
     if (response->has_ok_response()) {
       toAuthzResponseHeader(authz_response, response->ok_response().headers());
+      if (response->ok_response().headers_to_remove_size() > 0) {
+        for (const auto& header : response->ok_response().headers_to_remove()) {
+          authz_response->headers_to_remove.push_back(Http::LowerCaseString(header));
```

Comment:
Do you think we need to have a ban list of headers that can't be removed?

Reply:
That's a very good question.

I assume that since we are giving the authorization server permission to change headers that means the authorization server is trusted by Envoy. So we are not concerned that it may act maliciously. But that still leaves the possibility that the authorization server may change the headers in such a way as to make the request malformed.

Perhaps we should prevent it from removing the headers:
* :method
* :authority
* :path
* ... any others?

What do you think?

I just tested this manually and removing either of `:method` or `:authority` makes Envoy crash. `:path` does not.

Reply:
We could potentially also disallow removing:
* :protocol (could it be present in a request?)
* :scheme (could it be present in a request?)
* :status (should never be present in a request?)

Reply:
We could potentially also disallow removing:

* :protocol (could it be present in a request?)
* :scheme (could it be present in a request?)
* :status (should never be present in a request?)

Reply:
I found a relevant test for this:
https://github.com/envoyproxy/envoy/blob/master/test/common/router/config_impl_test.cc#L1723

Perhaps this is the list of headers we should disallow?

Reply:
Yeap. I think we should ban removing pseudo headers.

Reply:
Implemented + added a new test for this.